mlflow/pyfunc/__init__.py

```diff
@@ -480,7 +480,12 @@ def load_model(
 
     _add_code_from_conf_to_system_path(local_path, conf, code_key=CODE)
     data_path = os.path.join(local_path, conf[DATA]) if (DATA in conf) else local_path
-    model_impl = importlib.import_module(conf[MAIN])._load_pyfunc(data_path)
+    loader_module = conf[MAIN]
+    if loader_module in ["mlflow.tensorflow", "mlflow.keras"]:
+        model_type = mlflow.tensorflow._infer_model_type(model_meta)
+        model_impl = mlflow.tensorflow._load_pyfunc(data_path, model_type=model_type)
+    else:
+        model_impl = importlib.import_module(loader_module)._load_pyfunc(data_path)
```

Comment:
Can we move this logic into `mlflow.tensorflow._load_pyfunc`, as recommended here: https://github.com/mlflow/mlflow/pull/6530/files#r967536918?

Reply:
The logic need to use `model_meta` info, but `model_meta` is not passed to flavor's `_load_pyfunc` function.
If we force moving logic into `_load_pyfunc`, we need change `_load_pyfunc` interface arguments, (and other flavors we should update them as well), I think we can refactor it in following PR

Reply:
The `_load_pyfunc` method can load `model_meta` info from the directory `path` argument that is specified. I don't think we should make the change here.

Reply:
Updated :)